stats/cloud/api.go

```diff
@@ -89,40 +94,71 @@ func (c *Client) CreateTestRun(testRun *TestRun) (*CreateTestRunResponse, error)
 }
 
 func (c *Client) PushMetric(referenceID string, noCompress bool, samples []*Sample) error {
+	start := time.Now()
 	url := fmt.Sprintf("%s/metrics/%s", c.baseURL, referenceID)
 
-	if noCompress {
-		req, err := c.NewRequest("POST", url, samples)
-		if err != nil {
-			return err
-		}
-		return c.Do(req, nil)
+	jsonStart := time.Now()
+	b, err := json.Marshal(&samples)
+	if err != nil {
+		return err
 	}
+	jsonTime := time.Since(jsonStart)
 
-	var buf bytes.Buffer
-	var unzippedSize int
-	if samples != nil {
-		b, err := json.Marshal(&samples)
-		if err != nil {
-			return err
+	// TODO: change the context, maybe to one with a timeout
+	req, err := http.NewRequestWithContext(context.Background(), "POST", url, nil)
```

Comment:
Why "maybe"? :) A timeout would be good to have now.

Reply:
wanted to add it with 554c19e , but I wanted it set to the PushPeriod ... so we don't block forever, unfortunately that means more parameters, and this already has a global per client timeout of 20seconds ... so I will leave it for another time ...